### PR TITLE
Config improvements + fixed installation

### PR DESCRIPTION
**Pull request by sarveshmayil**
Added ability to change model parameters, loss arguments, etc.

Also fixed the conda + pip intallation setup, so should now work following instructions in README.md

Also removed a lot of deadcode, trimmed functions where possible

### CODE REVIEW

File: loss_functions.py [hunk 1]
```diff
@@ -155,48 +130,53 @@ def hypo_weight_loss(model_output):
...
+
+    if gt_normals is not None:
+        gt_normals = gt_normals.unsqueeze(0).float()
 
     coords = model_output["model_in"].float()
     pred_sdf = model_output["model_out"].float()
 
     sdf_constraint = torch.where(
-        gt_sdf == 0, torch.abs(pred_sdf), torch.zeros_like(pred_sdf)
+        torch.abs(gt_sdf) <= eps, torch.abs(pred_sdf), torch.zeros_like(pred_sdf)
```
Comment (yswi) [inline loss_functions.py]: Please revert this back to the original. We don't want to |s| < eps to be all 0.

File: loss_functions.py [hunk 2]
```diff
@@ -155,48 +130,53 @@ def hypo_weight_loss(model_output):
...
     )
+    losses['sdf'] = torch.abs(sdf_constraint).mean() * ks
 
     pred_sdf_c = torch.clip(pred_sdf, -0.3, 0.3).float()
     gt_sdf_c = torch.clip(gt_sdf, -0.3, 0.3).float()
 
     inter_constraint = torch.where(
-        gt_sdf == 0, torch.zeros_like(pred_sdf), abs(gt_sdf_c - pred_sdf_c)
+        torch.abs(gt_sdf) <= eps, torch.zeros_like(pred_sdf), abs(gt_sdf_c - pred_sdf_c)
```
Comment (yswi) [inline loss_functions.py]: This as well.

File: loss_functions.py [hunk 3]
```diff
@@ -155,48 +130,53 @@ def hypo_weight_loss(model_output):
...
     gradient = diff_operators.gradient(pred_sdf, coords)
 
-    norm = (1 - F.cosine_similarity(gradient, gt_normals, dim=-1))[..., None]
-    normal_constraint = torch.where(
-        gt_sdf == 0, norm, torch.zeros_like(gradient[..., :1])
-    )
+    if gt_normals is not None:
+        norm = (1 - F.cosine_similarity(gradient, gt_normals, dim=-1))[..., None]
+        normal_constraint = torch.where(
+            torch.abs(gt_sdf) <= eps, norm, torch.zeros_like(gradient[..., :1])
```
Comment (yswi) [inline loss_functions.py]: This as well. Please update this globally.

File: loss_functions.py [hunk 4]
```diff
@@ -155,48 +130,53 @@ def hypo_weight_loss(model_output):
     return weight_sum * (1 / total_weights)
 
 
-def hyper_loss(model_output, gt_sdf, gt_normals, ks, ki, kn, kg):
+def hyper_loss(model_output, gt_sdf, ks, ki, kg, gt_normals=None, kn=None, eps=0.0):
```
Comment (yswi) [inline loss_functions.py]: we don't want eps -- this is not what we have reported in the paper

File: utilities/sdf_meshing.py [hunk 5]
```diff
@@ -6,32 +6,33 @@
...
-import diff_operators
+from tqdm import tqdm
 
 
 def create_mesh(
     decoder,
     filename,
     N=256,
-    max_batch=40**3,
+    max_batch=32**3,
```
Comment (yswi) [inline utilities/sdf_meshing.py:18]: why ?

File: virdo.py [hunk 6]
```diff
@@ -8,12 +8,20 @@
...
 from utilities.sdf_meshing import create_mesh
-import modules, diff_operators, meta_modules
+import modules, meta_modules
 
-## Training Parameters
-lr = 3e-4
-epochs_til_ckpt = 5000
+from typing import List
 
+class LRscheduler:
```
Comment (yswi) [inline virdo.py]: Why do we need Learning rate scheduler? let's remove if not used.

File: virdo.py [hunk 7]
```diff
@@ -107,22 +123,25 @@ def from_pretraining(self, pretrained_path):
...
-    def pretraining(self, checkpoints_dir, end_epoch=10001):
-        make_dir(checkpoints_dir)
+    def pretraining(self, args):
+        make_dir(args['checkpoint_dir'])
         start_epoch = 0
         self.optims = torch.optim.Adam(
             [
-                {"params": self.object_code.parameters(), "lr": 1e-4},
-                {"params": self.object_model.parameters(), "lr": 1e-4},
+                {"params": self.object_code.parameters(), "lr": args['obj_code_lr_schedule']['initial']},
```
Comment (yswi) [inline virdo.py]: Let's not use learning rate scheduler if training goes well it :D

File: virdo.py [hunk 8]
```diff
@@ -231,52 +249,39 @@ def from_maintraining(self, main_trained_dir):
...
-                                                ).to(self.device)
-
         self.optims = torch.optim.Adam([
-            {'params': self.force_model.parameters(), 'lr': 5e-5},
-            {'params': self.deformation_model.parameters(), 'lr': 5e-5}
+            {'params': self.force_model.parameters(), 'lr': args['force_module_lr_schedule']['initial']},
+            {'params': self.deformation_model.parameters(), 'lr': args['def_module_lr_schedule']['initial']}
         ])
-
+        self._init_lr_schedulers((args['force_module_lr_schedule'], args['def_module_lr_schedule']))
```
Comment (yswi) [inline virdo.py]: Let's also remove the LRS if possible :)

**Review by yswi — APPROVED** [review]

**Comment by yswi** [comment]
Parametrization looks great
